validator/client/src/main/java/tech/pegasys/teku/validator/client/proposerconfig/FileProposerConfigProvider.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2022 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.client.proposerconfig;
+
+import java.io.File;
+import tech.pegasys.teku.infrastructure.async.AsyncRunner;
+import tech.pegasys.teku.validator.client.ProposerConfig;
+import tech.pegasys.teku.validator.client.proposerconfig.loader.ProposerConfigLoader;
+
+public class FileProposerConfigProvider extends AbstractProposerConfigProvider {
+  private final File source;
```

Comment:
Having specific subclasses for the config provider to handle URLs vs Files feels like overkill.  We could just use `File.toURL` to convert any file paths to URLs and then they could just be treated the same.